controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +61,393 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
 //+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacerequests/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacerequests/finalizers,verbs=update
 
-// Reconcile ensures that there is an SpaceRequest resource defined in the target member cluster
+// Reconcile ensures that there is a SpaceRequest resource defined in the target member cluster
 func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
 	logger.Info("reconciling SpaceRequest")
 
-	// TODO implement logic
+	// Fetch the SpaceRequest
+	// search on all member clusters
+	spaceRequest := &toolchainv1alpha1.SpaceRequest{}
+	var memberClusterWithSpaceRequest cluster.Cluster
+	var err error
+	for _, memberCluster := range r.MemberClusters {
+		err = memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: request.Namespace,
+			Name:      request.Name,
+		}, spaceRequest)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				// Error reading the object - requeue the request.
+				return reconcile.Result{}, errs.Wrap(err, "unable to get the current SpaceRequest")
+			}
+
+			//  spacerequest CR not found on current membercluster
+			continue
+		}
+
+		// save the member cluster on which the SpaceRequest CR was found
+		memberClusterWithSpaceRequest = memberCluster
+		break // exit once found
+	}
+	// if we exited with a notFound error
+	// it means that we couldn't find the spacerequest object on any of the given member clusters,
+	// let's return the error.
+	if err != nil && errors.IsNotFound(err) {
+		return reconcile.Result{}, errs.Wrap(err, "unable to find SpaceRequest")
+	}
+
+	if util.IsBeingDeleted(spaceRequest) {
+		logger.Info("spaceRequest is being deleted")
+		return reconcile.Result{}, r.ensureSpaceRequestDeletion(logger, memberClusterWithSpaceRequest, spaceRequest)
+	}
+	// Add the finalizer if it is not present
+	if err := r.addFinalizer(logger, memberClusterWithSpaceRequest, spaceRequest); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	subSpace, err := r.ensureSpace(logger, memberClusterWithSpaceRequest, spaceRequest)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update spaceRequest conditions and target cluster url
+	err = updateSpaceRequest(memberClusterWithSpaceRequest, spaceRequest, subSpace)
+
+	return ctrl.Result{}, err
+}
+
+// updateSpaceRequest updates conditions and target cluster url of the spaceRequest with the ones on the Space resource
+func updateSpaceRequest(memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
+	// reflect space conditions and target URL to spaceRequest conditions
+	spaceRequest.Status.Conditions = subSpace.Status.Conditions
+	spaceRequest.Status.TargetClusterURL = subSpace.Status.TargetCluster
```

Comment:
the `subSpace.Status.TargetCluster` contains the name of the member cluster, not the URL. You need to get the URL from ToolchainCluster CR.

Reply:
🤦‍♂️  My bad! thanks for pointing this out!

It should be fixed in [df7bfda](https://github.com/codeready-toolchain/host-operator/pull/770/commits/df7bfda7359f77a69b1b9522efa41c16b7b1eb30).

I've also added a check in [this commit](https://github.com/codeready-toolchain/toolchain-e2e/pull/685/commits/b8f1c40e114639246461c9b395886f4a90d7e607) of the paired pr in e2e tests to check that the APIEndpoint is being set. 

Please let me know if this fixes the issue.